WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java

```diff
@@ -0,0 +1,641 @@
+        findViewById(R.id.text_all_new).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showListFragment(PluginListType.NEW);
+            }
+        });
+
+        boolean isLandscape = DisplayUtils.isLandscape(this);
+        int displayWidth = DisplayUtils.getDisplayPixelWidth(this);
+        int maxRows = isLandscape ? 6 : 4;
+        mRowWidth = Math.round(displayWidth / (maxRows - 0.4f));
+        mIconSize = mRowWidth;
+
+        configureRecycler(mSitePluginsRecycler);
+        configureRecycler(mPopularPluginsRecycler);
+        configureRecycler(mNewPluginsRecycler);
+
+        mSitePlugins.addAll(mPluginStore.getSitePlugins(mSite));
+        refreshAllPlugins();
+
+        if (savedInstanceState == null) {
+            fetchAllPlugins();
+        }
+    }
+
+    private void configureRecycler(@NonNull RecyclerView recycler) {
+        recycler.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false));
+        recycler.setHasFixedSize(true);
+        recycler.setAdapter(new PluginBrowserAdapter(this));
+    }
+
+    @Override
+    protected void onDestroy() {
+        mDispatcher.unregister(this);
+        if (mSearchView != null) {
+            mSearchView.setOnQueryTextListener(null);
+        }
+        super.onDestroy();
+    }
+
+    @Override
+    public void onBackPressed() {
+        setTitle(R.string.plugins);
+        super.onBackPressed();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.search, menu);
+
+        mSearchMenuItem = menu.findItem(R.id.menu_search);
+        mSearchMenuItem.setOnActionExpandListener(this);
+        mSearchView = (SearchView) mSearchMenuItem.getActionView();
+        mSearchView.setOnQueryTextListener(this);
+
+        if (!TextUtils.isEmpty(mLastSearch)) {
+            mSearchMenuItem.expandActionView();
+            onQueryTextSubmit(mLastSearch);
+            mSearchView.setQuery(mLastSearch, true);
+        }
+
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            onBackPressed();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putSerializable(WordPress.SITE, mSite);
+        if (mSearchMenuItem != null && mSearchMenuItem.isActionViewExpanded()) {
+            outState.putString(KEY_LAST_SEARCH, mSearchView.getQuery().toString());
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RequestCodes.PLUGIN_DETAIL) {
```

Comment:
Unless we have really have to, we shouldn't start a new request for the site plugins after plugin detail page is closed. When a plugin is removed, our api is a little slow to reflect it in the response and if it's requested immediately after, it'll look like the removed plugin is still installed.

Neither the site plugins nor the plugin directory are very dynamic lists and any action taken for a site plugin will be reflected in the DB, so unless I am missing something, we don't really need to do it.

Reply:
This is only refreshing the plugins from the DB - it's not requesting them again from the backend.

Reply:
🤦‍♂️